extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -1010,40 +1011,73 @@ export class AmpAutocomplete extends AMP.BaseElement {
   /**
    * Writes the selected value into the input field.
    * @param {?Element} element
+   * @return {?string}
    * @private
    */
-  selectItem_(element) {
+  setInputValue_(element) {
     if (element === null || element.hasAttribute('data-disabled')) {
-      return;
+      return null;
     }
     const selectedValue =
       element.getAttribute('data-value') || element.textContent;
 
-    this.fireSelectEvent_(selectedValue);
-    this.clearAllItems_();
-    this.toggleResults_(false);
-
     this.inputElement_.value = this.binding_.getUserInputForUpdateWithSelection(
       selectedValue,
       this.inputElement_,
       this.userInput_
     );
     this.userInput_ = this.binding_.getUserInputForUpdate(this.inputElement_);
+
+    return selectedValue;
+  }
+
+  /**
+   * Finish item selection with the selected value.
+   * @param {?string} value
+   * @private
+   */
+  selectItem_(value) {
+    if (value === null) {
+      return;
+    }
+    this.fireSelectEvent_(value);
+    this.clearAllItems_();
+    this.toggleResults_(false);
   }
 
   /**
-   * Triggers a 'select' event with the given value as the value emitted.
+   * Triggers select event on amp-autocomplete element and change events (bind
+   * and native, for amp-script) on child input/textarea with the given value
+   * as the value emitted.
    * @param {string} value
    * @private
    */
   fireSelectEvent_(value) {
-    const name = 'select';
-    const selectEvent = createCustomEvent(
+    let name = 'select';
+    let event = createCustomEvent(
+      this.win,
+      `amp-autocomplete.${name}`,
+      /** @type {!JsonObject} */ ({value})
+    );
+    this.action_.trigger(this.element, name, event, ActionTrust.HIGH);
+
+    // Appease type checking
+    if (!this.inputElement_) {
+      return;
+    }
+
+    // Ensure on="change" is triggered for input
+    name = 'change';
+    event = createCustomEvent(
       this.win,
       `amp-autocomplete.${name}`,
       /** @type {!JsonObject} */ ({value})
     );
-    this.action_.trigger(this.element, name, selectEvent, ActionTrust.HIGH);
+    this.action_.trigger(this.inputElement_, name, event, ActionTrust.HIGH);
+
+    // Ensure native change listeners in user amp-scripts are triggered
```

Comment:
nit: `“Ensure native change listeners are triggered”` — Though this change makes it easier to use amp-script with amp-autocomplete, there is not a high-level guarantee that AMP components should integrate cleanly with amp-script.

Reply:
Please update the comment mentioned mentioned above. Thanks!

Reply:
Done

Reply:
Cool, thanks! I'll merge when the build is green. :)